xapian-applications/omega/omindextest

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+# omindextest: Test omindex
+#
+# Copyright (C) 2019 Bruno Baruffaldi
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+# USA
+
+DIR=$(pwd)
+TEST_FILES="$DIR/testfiles/"
+TEST_DB="$DIR/testdatabase"
+
+./omindex --db $TEST_DB --url $TEST_FILES $TEST_FILES
+./omindexcheck $TEST_FILES $TEST_DB > "omindextest.log"
+failed=$(echo "$?")
+
+rm -r $TEST_DB
+
+if [ $failed = 0 ] ; then
+    exit 0
+fi
+exit 1
```

Comment:
Simpler to `exit "$failed"` which also preserves the actual non-zero exit status of the command rather than squashing any non-zero exit status to `1`.

But there's a better way to achieve this using the `trap` shell built-in - you can specify a command to be run on shell exit with:

```
trap 'rm -rf "$TEST_DB"' 0
```

So do that just before running `omindex` and the database will always be removed for you.

If you want the database to be removed if the script is killed by a signal, you can list signal numbers too - the portable ones are apparently `1 2 13 15`.  E.g. `2` is `SIGINT` which means that the database is removed if you hit Ctrl+C mid test:

```
trap 'rm -rf "$TEST_DB"' 0 1 2 13 15
```

With the clean-up outsourced to `trap`, you can add `set -e` at the start of the script so that the script fails if a command exists with non-zero status, and then there's no need to check `$?` or explicitly invoke `exit` at all - the script will exit with the status of the failed command.  And as a bonus, it will now also exit right away if `omindex` fails.

Generally you probably want to use `set -e` for most shell scripts - if there are particular commands that are allowed to fail you can use `command || true` to allow that.

Reply:
Okay, I am going to add `trap` and `set`

Reply:
Okay, I am going to add `trap` and `set`